### PR TITLE
BAEL-5124 - Adding a module with examples of printing thread info in the log file using log4j

### PR DESCRIPTION
**Pull request by josephdsbr**

### CODE REVIEW

**Review by JavierPeris — COMMENTED** [review]
I think a better module name would be log4j2-thread-info.

File: logging-modules/log4j2/src/main/java/com/baeldung/logging/log4j2threadinfo/Log4j2ThreadInfo.java [hunk 1]
```diff
@@ -0,0 +1,18 @@
+package com.baeldung.logging.log4j2ThreadInfo;
```
Comment (JavierPeris) [inline logging-modules/log4j2/src/main/java/com/baeldung/logging/log4j2threadinfo/Log4j2ThreadInfo.java:1]: This package name is not standard compliant. It must be in lowercase.

File: logging-modules/log4j2/src/main/java/com/baeldung/logging/log4j2threadinfo/Log4j2ThreadInfo.java [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.util.stream.IntStream;
+
+public class Log4j2ThreadInfo {
+    private static final Logger LOGGER = LogManager.getLogger(Log4j2ThreadInfo.class);
```
Comment (JavierPeris) [inline logging-modules/log4j2/src/main/java/com/baeldung/logging/log4j2threadinfo/Log4j2ThreadInfo.java:9]: Although a final variable, this object reference is not immutable, that is, it´s not a constant and therefore it should not be in uppercase.

File: logging-modules/log4j2/src/main/resources/log4j2.xml [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+            %d{yyyy-MM-dd HH:mm:ss.SSS} --- thread_id="%tid" thread_name="%tn" thread_priority="%tp" --- [%p] %m%n
+        </Property>
+    </Properties>
+    <Appenders>
+        <RollingFile name="RollingName" filename="logs/app.log" filePattern="$${date:yyyy-MM}/app-%d{MM-dd-yyyy}-%i.log">
+            <PatternLayout>
+                <Pattern>${LOG_PATTERN}</Pattern>
+            </PatternLayout>
+            <Policies>
+                <SizeBasedTriggeringPolicy size="20MB" />
```
Comment (JavierPeris) [inline logging-modules/log4j2/src/main/resources/log4j2.xml:14]: Applying the formatter the spaces between value and closing field dissapears:
`<SizeBasedTriggeringPolicy size="20MB"/>`

File: logging-modules/log4j2/src/main/java/com/baeldung/logging/log4j2threadinfo/Log4j2ThreadInfo.java [hunk 4]
```diff
@@ -0,0 +1,18 @@
+package com.baeldung.logging.log4j2threadinfo;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.util.stream.IntStream;
```
Comment (JavierPeris) [inline logging-modules/log4j2/src/main/java/com/baeldung/logging/log4j2threadinfo/Log4j2ThreadInfo.java:6]: Applying the formatter swap the order of the import statements.

Comment (josephdsbr) [reply logging-modules/log4j2/src/main/java/com/baeldung/logging/log4j2threadinfo/Log4j2ThreadInfo.java:6]: I applied it, but nothing changed in the import order. I think it is already formatted.

File: logging-modules/log4j2/src/main/resources/log4j2.xml (hunk 3 above)
Comment (josephdsbr) [reply logging-modules/log4j2/src/main/resources/log4j2.xml]: Done

**Review by JavierPeris — APPROVED** [review]

**Pull request merged**